MaterialDesignThemes.Wpf/ScrollViewerAssist.cs

```diff
@@ -119,10 +124,24 @@ static void OnLoaded(ScrollViewer scrollViewer, Action<ScrollViewer> doOnLoaded)
             }
         }
 
+        static void OnUnloaded(ScrollViewer scrollViewer, Action<ScrollViewer> doOnUnloaded)
+        {
+            if (!scrollViewer.IsLoaded)
```

Comment:
Sorry for barging in on this code review; I was curious. 😃 

This guard is a bit suspicious to me. Why do we only want to listen for the Unloaded event if the `ScrollViewer` is not already loaded? I think the situation where it IS loaded is equally relevant. This would be a scenario where the AP is not statically set in XAML, but applied at runtime after the `ScrollViewer` has loaded. I acknowledge this may not be the most common use case, but it is still possible. In short, I think we should handle both the `ScrollViewer.IsLoaded=true/false` cases.

Another thing to consider is using weak event handlers. Both the `Loaded` and `Unloaded` event handlers are currently registered as strong event handlers which unregister themselves when the corresponding event fires. However, I think there are scenarios where this will not work - not particularly related to your changes though:

If the `ScrollViewer` resides in a `TabÌtem`, it will be loaded and unloaded multiple times when switching between the tabs. The current implementation only seems to wire up the hooks when the AP changes which means it will register for the `Loaded` event at startup. However, after navigating away from the `TabItem` and back to it again, the `Unloaded` event will fire, removing the hook, and then there is nothing listening for the "second" `Loaded` event to re-register the hook. Perhaps this could be mitigated by simply using weak event handlers and never unregistering them?

This comment also applies for the changes to the other AP.

Thoughts?

Reply:
The guard is somewhat pointless. As the handler is only fired after the `ScrollViewer` has been unloaded `scrollViewer.IsLoaded` will never be true. I'm happy to remove that guard to simplify the code.

As to your other points, weak event handlers are certainly worth considering.
As the code stands it handles the case when the tab is loaded and unloaded multiple times, as the `PropertyChangedCallback` methods are fired every time the control is loaded (and the properties are re-attached).
I have tested this in the demo app.